test/avocado/seleniumlib.py

```diff
@@ -95,30 +95,38 @@ def tearDown(self):
             raise Exception('ERR: Unable to close WEBdriver', str(e))
 
     def everything_loaded(self, element):
+        """
+This function is only for internal purposes:
+    It via javascript check that attribute data-loaded is in element
+        """
         if javascript_operations:
-            for foo in range(0, self.default_try): 
-                if self.driver.execute_script("return Array.from(arguments[0].children).every(function (e) { return e.getAttribute('data-loaded'); })", element):
-                    break
+            return self.driver.execute_script("return arguments[0].getAttribute('data-loaded')", element)
+        else:
+            return True
 
     def click(self, element):
-        notdone=""
-        self.everything_loaded(element)
+        failure = "CLICK: too many tries"
+        usedfunction = self.element_wait_functions[element]
         for foo in range(0, self.default_try):
             try:
                 if javascript_operations:
                     self.driver.execute_script("arguments[0].click();", element)
                 else:
                     element.click()
+                failure = None
                 break
-                notdone=None
             except Exception as e:
-                notdone=e
-                time.sleep(self.default_explicit_wait)
+                failure = e
                 pass
-        if notdone:
+            try:
+                element = self.usedfunction()
```

Comment:
not `self.` and it may not be set
maybe like this?
element = usedfunction() if usedfunction else element